ledger/participant-integration-api/src/main/scala/platform/store/dao/JdbcLedgerDao.scala

```diff
@@ -1204,9 +1252,109 @@ private[platform] object JdbcLedgerDao {
         |set referential_integrity true;
       """.stripMargin
 
+    override def limit(numberOfItems: Int): String = s"limit $numberOfItems"
+
     /** H2 does not support asynchronous commits */
     override protected[JdbcLedgerDao] def enforceSynchronousCommit(implicit
         conn: Connection
     ): Unit = ()
+
+    //H2 needs a backtick to escape reserved words, double-quote is problematic
+    override protected[JdbcLedgerDao] def escapeReservedWord(word: String): String =
+      s""" `$word` """.trim
+
+  }
+
+  object OracleQueries extends Queries {
+
+    override protected[JdbcLedgerDao] val SQL_INSERT_PACKAGE: String = {
+      """merge into packages p using (select ledger_end from parameters) par
+              |on (p.package_id = {package_id})
+              |when not matched then
+              |insert (package_id, upload_id, source_description, "size", known_since, ledger_offset, package)
+              |values ({package_id}, {upload_id}, {source_description}, {size}, {known_since}, par.ledger_end, {package})
+              |""".stripMargin
+    }
+
+    override protected[JdbcLedgerDao] val SQL_INSERT_COMMAND: String =
+      """merge into participant_command_submissions pcs
+        |using dual
+        |on (pcs.deduplication_key ={deduplicationKey})
+        |when matched then
+        |  update set pcs.deduplicate_until={deduplicateUntil}
+        |  where pcs.pcs.deduplicate_until < {submittedAt}
```

Comment:
Is `pcs.pcs` intentional?

```suggestion
        |  where pcs.deduplicate_until < {submittedAt}
```

Reply:
addressed